src/ui/components/Rating/index.js

```diff
@@ -46,6 +51,34 @@ export class RatingBase extends React.Component {
     // Accept falsey values as if they are zeroes.
     const rating = this.props.rating || 0;
 
+    // We render read-only star ratings with half-stars, to give users a
+    // better representation of the average rating.
+    if (readOnly) {
```

Comment:
I went through this a few times making it work with tests and trying to make it as concise as possible, but I dunno if it's that accessible. It's weird code that write. Happy to hear about ways it would be considered more accessible; it works but it's not lovely.